### ISSUE TITLE
[CI] IndexShardIT#testStressMaybeFlushOrRollTranslogGeneration failure

### ISSUE THREAD

**Issue by martijnvg** [elastic/elasticsearch#25773]
Build url: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+java9-periodic/3287/consoleFull

Reproduce line: `gradle :core:integTest -Dtests.seed=D51E246A7DC3F163 -Dtests.class=org.elasticsearch.index.shard.IndexShardIT -Dtests.security.manager=true -Dtests.jvm.argline="--add-opens=java.base/java.lang=ALL-UNNAMED --add-opens=java.base/java.nio=ALL-UNNAMED --add-opens=java.base/java.nio.file=ALL-UNNAMED --add-opens=java.base/java.util=ALL-UNNAMED --add-opens=java.base/java.util.concurrent=ALL-UNNAMED --add-opens=java.base/java.util.concurrent.locks=ALL-UNNAMED --add-opens=java.base/java.util.regex=ALL-UNNAMED" -Dtests.locale=en-US -Dtests.timezone=UTC` (does not reproduce for me)

```
FAILURE 10.1s J0 | IndexShardIT.testStressMaybeFlushOrRollTranslogGeneration <<< FAILURES!
   > Throwable #1: java.lang.AssertionError: expected:<1> but was:<2514855>
   > 	at __randomizedtesting.SeedInfo.seed([D51E246A7DC3F163:3812107F9AD40345]:0)
   > 	at org.elasticsearch.index.shard.IndexShardIT.lambda$testStressMaybeFlushOrRollTranslogGeneration$8(IndexShardIT.java:457)
   > 	at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:722)
   > 	at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:696)
   > 	at org.elasticsearch.index.shard.IndexShardIT.testStressMaybeFlushOrRollTranslogGeneration(IndexShardIT.java:465)
   > 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
   > 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
   > 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
   > 	at java.base/java.lang.reflect.Method.invoke(Method.java:564)
   > 	at java.base/java.lang.Thread.run(Thread.java:844)
   > 	Suppressed: java.lang.AssertionError: expected:<1> but was:<0>
   > 		at org.elasticsearch.index.shard.IndexShardIT.lambda$testStressMaybeFlushOrRollTranslogGeneration$8(IndexShardIT.java:457)
   > 		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:710)
   > 		... 38 more
   > 	Suppressed: java.lang.AssertionError: expected:<1> but was:<0>
   > 		at org.elasticsearch.index.shard.IndexShardIT.lambda$testStressMaybeFlushOrRollTranslogGeneration$8(IndexShardIT.java:457)
   > 		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:710)
   > 		... 38 more
   > 	Suppressed: java.lang.AssertionError: expected:<1> but was:<0>
   > 		at org.elasticsearch.index.shard.IndexShardIT.lambda$testStressMaybeFlushOrRollTranslogGeneration$8(IndexShardIT.java:457)
   > 		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:710)
   > 		... 38 more
   > 	Suppressed: java.lang.AssertionError: expected:<1> but was:<0>
```

**Comment by dnhatn** [comment]
I am looking at this.

**Comment by dnhatn** [comment]
The test was stuck in an infinite loop of flushing - it expected one more flush but the actual value was 247006 flushes.
```
> 	Suppressed: java.lang.AssertionError: expected:<1> but was:<247006>
> 		at org.elasticsearch.index.shard.IndexShardIT.lambda$testStressMaybeFlushOrRollTranslogGeneration$8(IndexShardIT.java:456)
> 		at org.elasticsearch.test.ESTestCase.assertBusy(ESTestCase.java:714)
```
This issue were addressed by #28350.
